build/Build.cs

```diff
@@ -0,0 +1,195 @@
+    ///   - JetBrains Rider            https://nuke.build/rider
+    ///   - Microsoft VisualStudio     https://nuke.build/visualstudio
+    ///   - Microsoft VSCode           https://nuke.build/vscode
+
+    public static int Main() => Execute<Build>(x => x.CreatePackage);
+
+    [Parameter("Configuration to build - Default is 'Debug' (local) or 'Release' (server)")]
+    readonly Configuration Configuration = IsLocalBuild ? Configuration.Debug : Configuration.Release;
+
+    [Solution] readonly Solution Solution;
+    [GitRepository] readonly GitRepository GitRepository;
+    [GitVersion] readonly GitVersion GitVersion;
+
+    [Parameter] string NugetPrerelease;
+
+    // Directories
+    AbsolutePath ToolsDir => RootDirectory / "tools";
+    AbsolutePath Output => RootDirectory / "bin";
+    AbsolutePath OutputNuget => Output / "nuget";
+    AbsolutePath OutputTests => RootDirectory / "TestResults";
+    AbsolutePath OutputPerfTests => RootDirectory / "PerfResults";
+    AbsolutePath SourceDirectory => RootDirectory / "src";
+    AbsolutePath DocSiteDirectory => RootDirectory / "docs/_site";
+    AbsolutePath ArtifactsDirectory => RootDirectory / "artifacts";
+    public string ChangelogFile => RootDirectory / "CHANGELOG.md";
+
+    [Parameter] readonly string Source = "https://resharper-plugins.jetbrains.com/api/v2/package";
+
+    static readonly JsonElement? _githubContext = string.IsNullOrWhiteSpace(EnvironmentInfo.GetVariable<string>("GITHUB_CONTEXT")) ? 
+        null 
+        : JsonSerializer.Deserialize<JsonElement>(EnvironmentInfo.GetVariable<string>("GITHUB_CONTEXT"));
+
+    //let hasTeamCity = (not (buildNumber = "0")) // check if we have the TeamCity environment variable for build # set
+    static readonly int BuildNumber = _githubContext.HasValue ? int.Parse(_githubContext.Value.GetProperty("run_number").GetString()) : 0;
+        
+    public ChangeLog Changelog => ReadChangelog(ChangelogFile);
+
+    public ReleaseNotes LatestVersion => Changelog.ReleaseNotes.OrderByDescending(s => s.Version).FirstOrDefault() ?? throw new ArgumentException("Bad Changelog File. Version Should Exist");
+    public string ReleaseVersion => LatestVersion.Version?.ToString() ?? throw new ArgumentException("Bad Changelog File. Define at least one version");
+
+    Target Clean => _ => _
+        .Before(Restore)
+        .Executes(() =>
+        {            
+            SourceDirectory
+            .GlobDirectories("**/bin", "**/obj", Output, OutputTests, OutputPerfTests, OutputNuget, DocSiteDirectory)
+            .ForEach(DeleteDirectory);
+            EnsureCleanDirectory(ArtifactsDirectory);
+        });
+
+    Target Restore => _ => _
+        .Executes(() =>
+        {
+            DotNetRestore(s => s
+                .SetProjectFile(Solution));
+        });
+    IEnumerable<string> ChangelogSectionNotes => ExtractChangelogSectionNotes(ChangelogFile);
+
+    Target RunChangelog => _ => _
```

Comment:
Finalizes the changelog by moving all entries from `[vNext]` to the version specified by release.